src/test/java/org/owasp/esapi/logging/appender/LogPrefixAppenderTest.java

```diff
@@ -22,96 +21,146 @@
+        whenNew(ClientInfoSupplier.class).withNoArguments().thenReturn(cisSpy);
+        whenNew(ServerInfoSupplier.class).withArguments(testLoggerName).thenReturn(sisSpy);
+
+        LogPrefixAppender lpa = new LogPrefixAppender(false, false, false, false, null);
+        lpa.appendTo(testLoggerName, testEventType, testLogMessage);
+
+        verify(uisSpy, times(1)).setLogUserInfo(false);
+        verify(cisSpy, times(1)).setLogClientInfo(false);
+        verify(sisSpy, times(1)).setLogServerIp(false);
+        verify(sisSpy, times(1)).setLogApplicationName(false, null);
+    }
+    
+    @Test
+    public void testDelegateCtrArgs() throws Exception {
+        ArgumentCaptor<EventType> eventTypeCapture = ArgumentCaptor.forClass(EventType.class);
+        ArgumentCaptor<String> logNameCapture = ArgumentCaptor.forClass(String.class);
+        whenNew(EventTypeLogSupplier.class).withArguments(eventTypeCapture.capture()).thenReturn(etlsSpy);
+        whenNew(UserInfoSupplier.class).withNoArguments().thenReturn(uisSpy);
+        whenNew(ClientInfoSupplier.class).withNoArguments().thenReturn(cisSpy);
+        whenNew(ServerInfoSupplier.class).withArguments(logNameCapture.capture()).thenReturn(sisSpy);
+
+        LogPrefixAppender lpa = new LogPrefixAppender(true, true,true,true, testApplicationName);
+        lpa.appendTo(testLoggerName, testEventType, testLogMessage);
+
+        assertEquals(testEventType, eventTypeCapture.getValue());
+        assertEquals(testLoggerName, logNameCapture.getValue());
+    }
+
+    @Test
+    public void testLogContentWhenClientInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, UIS_RESULT, EMPTY_RESULT,SIS_RESULT, "[EVENT_TYPE USER_INFO -> SERVER_INFO]");
+    }
+
+
+    @Test
+    public void testLogContentWhenUserInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, EMPTY_RESULT, CIS_RESULT,SIS_RESULT, "[EVENT_TYPE CLIENT_INFO -> SERVER_INFO]");
+    }
+    
+    @Test
+    public void testLogContentWhenClientInfoEmptyAndServerInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, UIS_RESULT, EMPTY_RESULT,EMPTY_RESULT, "[EVENT_TYPE USER_INFO]");
+    }
+
+    @Test
+    public void testLogContentWhenUserInfoEmptyAndServerInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, EMPTY_RESULT, CIS_RESULT,EMPTY_RESULT, "[EVENT_TYPE CLIENT_INFO]");
+    }
+
+    @Test
+    public void testLogContentWhenUserInfoAndClientInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, EMPTY_RESULT, EMPTY_RESULT, SIS_RESULT, "[EVENT_TYPE -> SERVER_INFO]");
+    }
+
+    @Test
+    public void testLogContentWhenServerInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, UIS_RESULT, CIS_RESULT, EMPTY_RESULT, "[EVENT_TYPE USER_INFO:CLIENT_INFO]");
+    }
+    
+    @Test
+    public void testLogContentWhenUserInfoEmptyAndClientInfoEmptyAndServerInfoEmpty() throws Exception {
+        runTest(ETL_RESULT, EMPTY_RESULT, EMPTY_RESULT, EMPTY_RESULT, "[EVENT_TYPE]");
+    }
```

Comment:
@ricksouza @cristiantm
I believe these are all the pertinent test cases that were previously inconsistent.

Reply:
https://github.com/ESAPI/esapi-java-legacy/pull/529#issuecomment-571124457